### PR TITLE
Update packages

### PR DESCRIPTION
**Pull request by ChadNedzlek**
Update to .NET 6.0 and use some newer API's

### CODE REVIEW

File: src/DotNet.Status.Web.Tests/TimeLineIssueTriageTests.cs [hunk 1]
```diff
@@ -5,6 +5,13 @@
 using System.Collections.Generic;
 using System.IO;
 using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.Data.OData.Query.SemanticAst;
+using Microsoft.DotNet.GitHub.Authentication;
+using Microsoft.DotNet.Services.Utility;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Internal;
+using Octokit;
```
Comment (alexperovich) [inline src/DotNet.Status.Web.Tests/TimeLineIssueTriageTests.cs:14]: Why add a bunch of new usings?

File: Directory.Build.props [hunk 2]
```diff
@@ -12,8 +12,21 @@
...
   <PropertyGroup>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
-    <NoWarn>$(NoWarn);NU5125</NoWarn>
+    <TargetFramework>net6.0</TargetFramework>
+    <NoWarn>$(NoWarn);NU5125;CA1416</NoWarn>
+    <!--
+      We sort of half sort of support macOS in this repo, this warning is telling us that we are on the precipice of 
+      failure (because we are depending on win-x64 only assemblies). But darc is basically standing in the middle
+      of a thunderstorm and just hasn't been struck by lightning.
+      We don't have any validation around this scenario, and we dont' have time to separate out all our dependencies
```
Comment (adiaaida) [inline Directory.Build.props:21]:
```suggestion
      We don't have any validation around this scenario, and we don't have time to separate out all our dependencies
```

File: Directory.Build.props [hunk 3]
```diff
@@ -12,8 +12,21 @@
   </PropertyGroup>
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
-    <NoWarn>$(NoWarn);NU5125</NoWarn>
+    <TargetFramework>net6.0</TargetFramework>
+    <NoWarn>$(NoWarn);NU5125;CA1416</NoWarn>
+    <!--
+      We sort of half sort of support macOS in this repo, this warning is telling us that we are on the precipice of
```
Comment (adiaaida) [inline Directory.Build.props:18]: Opened: https://github.com/dotnet/arcade/issues/9192

We should probably reference it here.

File: src/Maestro/MaestroApplication/ApplicationPackageRoot/ApplicationManifest.xml [hunk 4]
```diff
@@ -132,6 +132,6 @@
     <DefaultRunAsPolicy UserRef="MaestroUser" />
   </Policies>
   <Certificates>
-    <EndpointCertificate X509FindType="FindBySubjectName" X509FindValue="[WebCertName]" Name="WebSslCert" />
+    <EndpointCertificate X509FindValue="[WebCertName]" Name="WebSslCert" />
```
Comment (alexperovich) [inline src/Maestro/MaestroApplication/ApplicationPackageRoot/ApplicationManifest.xml:135]: I see VS is still removing this. We should probably put it back.

File: global.json [hunk 5]
```diff
@@ -1,13 +1,13 @@
 {
   "sdk": {
-    "version": "5.0.406",
+    "version": "6.0.202",
```
Comment (adiaaida) [inline global.json:3]: I don't know which arcade version flows into arcade-services. Should we try to update the subscription to the release/6.0 branch? @mmitche @riarenas

File: src/RolloutScorer/RolloutScorer/Commands.cs [hunk 6]
```diff
@@ -69,7 +69,7 @@ public async Task<int> InvokeAsync(IEnumerable<string> arguments)
             _rolloutScorer.GithubConfig = StandardConfig.DefaultConfig.GithubConfig;
 
             // If they haven't told us to upload but they also haven't specified a repo & rollout start date, we need to throw
-            if (string.IsNullOrEmpty(_rolloutScorer.Repo) || (_rolloutScorer.RolloutStartDate == null))
+            if (string.IsNullOrEmpty(_rolloutScorer.Repo))
```
Comment (alexperovich) [inline src/RolloutScorer/RolloutScorer/Commands.cs:72]: Did you remove this on purpose?

File: global.json (hunk 5 above)
Comment (adiaaida) [reply global.json:3]: (after this PR goes in, of course)

File: src/Microsoft.DncEng.Configuration.Extensions/Microsoft.DncEng.Configuration.Extensions.csproj [hunk 7]
```diff
@@ -2,7 +2,6 @@
 
   <PropertyGroup>
     <IsPackable>true</IsPackable>
-    <Nullable>enable</Nullable>
```
Comment (garath) [inline src/Microsoft.DncEng.Configuration.Extensions/Microsoft.DncEng.Configuration.Extensions.csproj:5]: Why not keep this?

File: src/Maestro/MaestroApplication/ApplicationPackageRoot/ApplicationManifest.xml [hunk 8]
```diff
@@ -132,6 +132,6 @@
     <DefaultRunAsPolicy UserRef="MaestroUser" />
   </Policies>
   <Certificates>
-    <EndpointCertificate X509FindType="FindBySubjectName" X509FindValue="[WebCertName]" Name="WebSslCert" />
```
Comment (adiaaida) [inline src/Maestro/MaestroApplication/ApplicationPackageRoot/ApplicationManifest.xml:135]: was this change intentional?

File: src/RolloutScorer/RolloutScorer/Commands.cs [hunk 9]
```diff
@@ -69,7 +69,7 @@ public async Task<int> InvokeAsync(IEnumerable<string> arguments)
             _rolloutScorer.GithubConfig = StandardConfig.DefaultConfig.GithubConfig;
 
             // If they haven't told us to upload but they also haven't specified a repo & rollout start date, we need to throw
-            if (string.IsNullOrEmpty(_rolloutScorer.Repo) || (_rolloutScorer.RolloutStartDate == null))
```
Comment (adiaaida) [inline src/RolloutScorer/RolloutScorer/Commands.cs:72]: @jonfortescue Is this change ok?

File: src/Shared/Microsoft.DotNet.GitHub.Authentication/GitHubAppTokenProvider.cs [hunk 10]
```diff
@@ -2,55 +2,53 @@
...
         public string GetAppToken()
         {
-            return GetAppToken(Options.DefaultName);
+            var options = _options.CurrentValue;
+            return GetAppToken(options.GitHubAppId, options.PrivateKey);
         }
-
-        /// <summary>
-        /// Get an app token using the <see cref="GitHubTokenProviderOptions"/> corresponding to the specified <see href="https://docs.microsoft.com/en-us/dotnet/core/extensions/options#named-options-support-using-iconfigurenamedoptions">named option</see>.
-        /// </summary>
```
Comment (garath) [inline src/Shared/Microsoft.DotNet.GitHub.Authentication/GitHubAppTokenProvider.cs:27]: Comment is still valid, why remove it?

File: src/RolloutScorer/RolloutScorer/Commands.cs (hunk 9 above)
Comment (jonfortescue) [reply src/RolloutScorer/RolloutScorer/Commands.cs:72]: No, the second conditional us necessary for checking if a rollout start date hasn't been provided

File: src/DotNet.Status.Web.Tests/TimeLineIssueTriageTests.cs (hunk 1 above)
Comment (ChadNedzlek) [reply src/DotNet.Status.Web.Tests/TimeLineIssueTriageTests.cs:14]: Ah, this was where my evil test was. I'll clean them up (probably jsut revert this file)

File: Directory.Build.props (hunk 3 above)
Comment (ChadNedzlek) [reply Directory.Build.props:18]: I think we probably want to do more than just that one project.  The x64/any disparity is all over the repo, we really need a solid plan to the dependency web here rather than "just hope we don't mess it up again". :-)

File: src/Maestro/MaestroApplication/ApplicationPackageRoot/ApplicationManifest.xml (hunk 4 above)
Comment (ChadNedzlek) [reply src/Maestro/MaestroApplication/ApplicationPackageRoot/ApplicationManifest.xml:135]: Damn it. :-|

File: global.json (hunk 5 above)
Comment (ChadNedzlek) [reply global.json:3]: No version flows into here (or into helix-services), to my knowledge. Azure services and arcade aren't... really friends. Arcade is really focused on products that release big stable packages once in a blue moon, not services that ship via special deployment instructions in code and don't actually produce any binaries... it's been a struggle.

File: src/Microsoft.DncEng.Configuration.Extensions/Microsoft.DncEng.Configuration.Extensions.csproj (hunk 7 above)
Comment (ChadNedzlek) [reply src/Microsoft.DncEng.Configuration.Extensions/Microsoft.DncEng.Configuration.Extensions.csproj:5]: Because there are a bunch of warnings I couldn't remove. You really can't add nullable to the _middle_ of a tech stack, because stuff below it can't be annotated so you end up in irreconcilable situations.  If we want to enable it, we shouldn't do it piecemeal in random projects, and rather have a real plan (which we can't afford)

File: src/RolloutScorer/RolloutScorer/Commands.cs (hunk 9 above)
Comment (ChadNedzlek) [reply src/RolloutScorer/RolloutScorer/Commands.cs:72]: It is not, because RolloutStartDate is a DateTimeOffset, which cannot be null.  It was a warning from the compiler that this code was pointless, so I removed it to fix the warning.

File: src/Shared/Microsoft.DotNet.GitHub.Authentication/GitHubAppTokenProvider.cs (hunk 10 above)
Comment (ChadNedzlek) [reply src/Shared/Microsoft.DotNet.GitHub.Authentication/GitHubAppTokenProvider.cs:27]: I didn't mean to.  There was a HUGE merge conflict in this file, and I must have resolved it badly. I'll put it back. :-)

Comment (ChadNedzlek) [reply src/Shared/Microsoft.DotNet.GitHub.Authentication/GitHubAppTokenProvider.cs:27]: (Though it's also a little useless to comment the provider, and not the service, because no one ever has a reference to the provider, only the service, so the comments aren't really visible... and it doesn't matter what _this_ class says it's does, only what the interface is designed to do. :-)

Comment (ChadNedzlek) [reply src/Shared/Microsoft.DotNet.GitHub.Authentication/GitHubAppTokenProvider.cs:27]: Put it back. :-)

File: src/RolloutScorer/RolloutScorer/Commands.cs (hunk 6 above)
Comment (ChadNedzlek) [reply src/RolloutScorer/RolloutScorer/Commands.cs:72]: I did. RolloutStartDate is a DateTimeOffset, and, therefor, never null.

**Comment by ChadNedzlek** [comment]
I think I address all the comments?

**Comment by ChadNedzlek** [comment]
I believe we are also _warning_ clean after this. I might turn on warnaserror, unless anyone objects?

**Comment by adiaaida** [comment]
Looks like the build still failed? Looks like Microsoft.DncEng.CommandLineLib didn't get converted

**Comment by ChadNedzlek** [comment]
Removed it's explicit \<TargetFramework\> that it had for some reason. :-)

File: src/RolloutScorer/RolloutScorer/Commands.cs (hunk 9 above)
Comment (adiaaida) [reply src/RolloutScorer/RolloutScorer/Commands.cs:72]: Is there a value that is the default for when a DateTimeOffset isn't set?

Comment (adiaaida) [reply src/RolloutScorer/RolloutScorer/Commands.cs:72]: I think the appropriate change here is to compare to DateTimeOffset.MinValue

File: src/Microsoft.DncEng.Configuration.Extensions/Microsoft.DncEng.Configuration.Extensions.csproj (hunk 7 above)
Comment (garath) [reply src/Microsoft.DncEng.Configuration.Extensions/Microsoft.DncEng.Configuration.Extensions.csproj:5]: Isn't it possible to do a slow adoption with the `#nullable` directive? These classes are already annotated, and I'd hate to lose that knowledge.

File: src/RolloutScorer/RolloutScorer/Commands.cs (hunk 9 above)
Comment (garath) [reply src/RolloutScorer/RolloutScorer/Commands.cs:72]: Would `_rolloutScorer.RolloutStartDate == default` also work?

File: src/Shared/Microsoft.DotNet.GitHub.Authentication/GitHubAppTokenProvider.cs (hunk 10 above)
Comment (garath) [reply src/Shared/Microsoft.DotNet.GitHub.Authentication/GitHubAppTokenProvider.cs:27]: Yeah, I could have been more thorough in adding the comments. But every bit helps, right? 😄

File: src/RolloutScorer/RolloutScorer/Commands.cs (hunk 9 above)
Comment (ChadNedzlek) [reply src/RolloutScorer/RolloutScorer/Commands.cs:72]: I'd rather not change existing behavior.  The existing behavior is as I left the code (it never could return true, so I removed the comparison).  I'd rather not mix a bunch of bug fixes into this already sadly huge change. :-(

File: src/Microsoft.DncEng.Configuration.Extensions/Microsoft.DncEng.Configuration.Extensions.csproj (hunk 7 above)
Comment (ChadNedzlek) [reply src/Microsoft.DncEng.Configuration.Extensions/Microsoft.DncEng.Configuration.Extensions.csproj:5]: The problem is they are incorrectly annotated.  You can slowly do it, but you have to start at the rootmost assembly (so that changes in base assemblies don't produce irreconcilable errors in assemblies that depend on them).  In my opinion, nullable costs more than is save anyway.

It also needs to be a plan to do it everywhere. Having to check which "kind" of assembly you are in, and having to check if you are supposed to slap "?" and "!" all over the place as you navigate around the project costs, me at least, a ton of time and frustration for the limit benefit it provides.  That's fine as a temporary state if the end goal is to use it everywhere, but I don't think the value is there for us to do that, and the costs of having a split code base are higher than just not having it anywhere.

File: src/RolloutScorer/RolloutScorer/Commands.cs (hunk 9 above)
Comment (adiaaida) [reply src/RolloutScorer/RolloutScorer/Commands.cs:72]: I hear that, but the comment suggests that the intention was to do something if the start date wasn't set. It's a bug and the code doesn't match the comment. So, IMO, you should either change it and remove the comment/change the comment so we go back and fix it, or just change it so it matches the comment and does what was intended.

Comment (ChadNedzlek) [reply src/RolloutScorer/RolloutScorer/Commands.cs:72]: I'll delete the comment then.

Comment (ChadNedzlek) [reply src/RolloutScorer/RolloutScorer/Commands.cs:72]: We shouldn't just fix all bugs in every PR... especially PR's that are, ostensibly, not intended to have any behavioral changes. I'm not saying there _isn't_ a bug... I'm just saying this PR isn't the appropriate place to fix said bug.  I'll open an issue to deal with it.

Comment (ChadNedzlek) [reply src/RolloutScorer/RolloutScorer/Commands.cs:72]: Created https://github.com/dotnet/arcade/issues/9206

File: src/Microsoft.DncEng.Configuration.Extensions/Microsoft.DncEng.Configuration.Extensions.csproj (hunk 7 above)
Comment (ChadNedzlek) [reply src/Microsoft.DncEng.Configuration.Extensions/Microsoft.DncEng.Configuration.Extensions.csproj:5]: This one actually looks like it was a base-most assembly... but it wasn't annotated correctly. So it wasn't a bad place to start... but we do need to commit to continuing (and actually do it correctly in the project we enable it for... not just enable it but leave all the violations).  Turning on nullable... but then disabling all the warnings it produces seems... counterproductive?

Comment (garath) [reply src/Microsoft.DncEng.Configuration.Extensions/Microsoft.DncEng.Configuration.Extensions.csproj:5]: That all makes sense.

**Comment by ChadNedzlek** [comment]
@alexperovich, the SQL migration went _really_ bad.  It has a legitmate complaint, our index IS (and, unfortunately, has always been) too long. We use 2 nvarchar(450) columns, which is 1800, the maximum index is 900, so we are just begging for our server to explode one day when someone adds a long name.

But, more importantly, there appears to be some sort of nullability constraint mismatch, and the history stuff isn't working.

**Comment by ChadNedzlek** [comment]
Looks like it builds and passes tests.  Anyone want to sign off? Or are there more issues that are causing a problem?

**Review by garath — APPROVED** [review]
